### PR TITLE
Database Backup

### PR DESCRIPTION
**Pull request by BryanGK**
This PR brings in spilo over cas-postgres due to the new spilo image having backups built-in.

### CODE REVIEW

File: helm/values.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ image:
     tag: latest # The tag value must be passed in via the deploy script
     clientSecret: "" # clientSecret must be passed in via deploy script
   psql:
-    repository: gcr.io/ggl-cas-storage/cas-postgres
+    repository: https://bcgov.github.io/spilo-chart
```
Comment (wenzowski) [inline helm/values.yaml:13]: this doesn't look like the url for a container image, is it?

**Review by wenzowski — COMMENTED** [review]
Did you get the spilo chart deployed before moving on to modifications of the psql init container?

File: helm/values.yaml (hunk 1 above)
Comment (BryanGK) [reply helm/values.yaml:13]: No, i suppose not. Where would that be found for spilo?

Comment (BryanGK) [reply helm/values.yaml:13]: isn't spilo the init container?

Comment (wenzowski) [reply helm/values.yaml:13]: Why are you changing it? Did the init container image for the `psql` client break?

Comment (wenzowski) [reply helm/values.yaml:13]: Ideally we'd reuse images wherever possible, but we also work in small increments wherever possible. If the psql client didn't break when moving to spilo chart as provider for patroni cluster then let's change as little as possible in this PR

File: helm/values.yaml [hunk 2]
```diff
@@ -40,32 +40,15 @@ deployer:
...
-      gcsBucket: ~
-      # kubernetesSecret should be gcp-{{ namespace }}-{{ gcs.bucketSuffix }}-service-account-key
-      kubernetesSecret: ~
-    # options below are required if walE.enable is true
-  namespace: ~
-  gcs:
-    bucketSuffix: ~
+spilo:
+  resources:
+  # default values spec'd from spilo-chart repo
```
Comment (matthieu-foucault) [inline helm/values.yaml:45]: The spilo-chart repo does say to adjust the resources as necessary, so I would keep the ones you had above

File: helm/values.yaml [hunk 3]
```diff
@@ -40,32 +40,15 @@ deployer:
...
-  patroni:
-    resources:
-      limits:
-        cpu: "210m"
-        memory: "512Mi"
-      requests:
-        cpu: "70m"
-        memory: "256Mi"
-    persistentVolume:
-      size: "300Mi"
```
Comment (matthieu-foucault) [inline helm/values.yaml:53]: you still need to define your PVC (and backup PVC) size with the spilo chart

File: helm/values.yaml (hunk 1 above)
Comment (matthieu-foucault) [reply helm/values.yaml:13]: There is no obligation to use the same image for the database and the init container. This is merely a convenience (don't need to create a separate dockefile just for `psql`) and optimization (the image will be more likely to be downloaded on the node already, if we use it for the database)

File: helm/templates/deploy.yaml [hunk 4]
```diff
@@ -63,11 +63,11 @@ spec:
         image: {{ .Values.image.psql.repository }}:{{ .Values.image.psql.tag }}
         resources:
           limits:
-            cpu: 50m
-            memory: 64Mi
+            cpu: {{ .Values.spilo.resources.limits.cpu }}
+            memory: {{ .Values.spilo.resources.limits.memory }}
```
Comment (matthieu-foucault) [inline helm/templates/deploy.yaml:67]: You don't need/want to change any of this. The containers running psql need way less resources than the ones running the actual database

File: helm/Chart.yaml [hunk 5]
```diff
@@ -5,6 +5,6 @@ type: application
 version: 0.1.0
 appVersion: "0.1.0"
 dependencies:
-  - name: cas-postgres
-    version: "0.8.3"
-    repository: https://bcgov.github.io/cas-postgres/
+  - name: spilo
+    version: 0.1.0
```
Comment (matthieu-foucault) [inline helm/Chart.yaml:9]:
```suggestion
    version: 0.2.0
```
Don't forget to run `helm dep up` after updating to update the `Chart.lock`

File: helm/templates/deploy.yaml (hunk 4 above)
Comment (BryanGK) [reply helm/templates/deploy.yaml:67]: If I find that the init container run painfully slow and I have resources, can I bring these numbers up a bit?

**Review by wenzowski — APPROVED** [review]
:shipit:

**Pull request merged**